xray/src/bin/inpaint_xray_quadtree.rs

```diff
@@ -0,0 +1,158 @@
+use structopt::StructOpt;
+use xray::{
+    generation::{assign_background_color, create_non_leaf_nodes, TileBackgroundColorArgument},
+    inpaint::perform_inpainting,
+    utils::{get_image_path, get_meta_pb_path},
+    Meta,
+};
+
+#[derive(StructOpt, Debug)]
+#[structopt(name = "inpaint_xray_quadtrees")]
+/// Inpaint a (possibly partial) xray quadtree. We assume that the input
+/// quadtree was generated with transparent background color, so we can
+/// determine which pixels to actually fill.
+struct CommandlineArguments {
+    /// Directory with the (possibly partial) quadtree to be inpainted.
+    /// Needs to include all leaf nodes of the neighboring quadtrees as well
+    /// for smooth inpainting results.
+    #[structopt(parse(from_os_str))]
+    input_directory: PathBuf,
+    /// Directory where to write the inpainted quadtree. Does *not*
+    /// have to be disjoint from input_directory.
+    #[structopt(parse(from_os_str), long)]
+    output_directory: PathBuf,
+    /// Tile background color.
+    #[structopt(default_value = "white", long)]
+    tile_background_color: TileBackgroundColorArgument,
+    /// The inpainting distance in pixels to fill holes (in particular useful for high resolutions)
+    #[structopt(long)]
+    inpaint_distance_px: u8,
+    /// The root node id to start inpainting with.
+    #[structopt(default_value = "r", long)]
+    root_node_id: NodeId,
+}
+
+fn get_adjacent_leaf_node_ids(
+    leaf_node_ids: &FnvHashSet<NodeId>,
+    input_directory: &Path,
+    root_node_id: NodeId,
+) -> FnvHashSet<NodeId> {
+    let spatial_root_node_id = SpatialNodeId::from(root_node_id);
+    let mut neighboring_leaf_node_ids = FnvHashSet::default();
+    let directions = &[
+        Direction::Left,
+        Direction::Top,
+        Direction::Right,
+        Direction::Bottom,
+    ];
+    for direction in directions {
+        if let Some(spatial_root_neighbor_id) = spatial_root_node_id.neighbor(*direction) {
+            if let Ok(neighbor_meta) = Meta::from_disk(get_meta_pb_path(
+                input_directory,
+                NodeId::from(spatial_root_neighbor_id),
+            )) {
+                for neighbor_node_id in neighbor_meta
+                    .nodes
+                    .iter()
+                    .filter(|id| id.level() == neighbor_meta.deepest_level)
+                {
+                    if let Some(spatial_id) =
+                        SpatialNodeId::from(*neighbor_node_id).neighbor(direction.opposite())
+                    {
+                        if leaf_node_ids.contains(&NodeId::from(spatial_id)) {
+                            neighboring_leaf_node_ids.insert(*neighbor_node_id);
+                        }
+                    }
+                }
+            }
+        }
+    }
+    neighboring_leaf_node_ids
+}
+
+fn copy_nodes(
+    node_ids: &FnvHashSet<NodeId>,
+    input_directory: &Path,
+    output_directory: &Path,
+) -> io::Result<()> {
+    for node_id in node_ids.iter() {
```

Comment:
can we use `try_for_each` here?

Reply:
Yes, done, but it doesn't make it much nicer, since the result of `try_for_each` wants a `()`

Reply:
Oh, that's too bad :\ I don't know which version is better, up to you.